lib/compress.js

```diff
@@ -1117,10 +1134,11 @@ merge(Compressor.prototype, {
             throw def;
         });
         def(AST_Dot, function(compressor){
-            if (compressor.option("unsafe") && this.property == "length") {
-                var str = ev(this.expression, compressor);
-                if (typeof str == "string")
-                    return str.length;
+            if (compressor.option("unsafe")) {
+                var val = ev(this.expression, compressor);
+                if (HOP(val, this.property)) {
```

Comment:
It just shows that hasOwnProperty is implemented differently on different node versions or platforms.

I suggest something like this:
```
if (val instanceof AST_Object && HOP(val, this.property)) {
```